src/main/resources/org/jenkinsci/plugins/badge/BadgeAction/index.groovy

```diff
@@ -30,44 +30,79 @@ l.layout {
 </style>
 """)
 
-        def base =  "${app.rootUrl}${my.project.url}";
-        def badge = base + "badge/icon"
-        
         def fullJobName = h.escape(my.project.fullName);
-        def publicbadge = "${app.rootUrl}buildStatus/icon?job=${fullJobName}";
+        def jobUrlWithView =  "${app.rootUrl}${my.project.url}";
+        def jobUrlWithoutView =  "${app.rootUrl}job/${fullJobName}";
```

Comment:
I suggest that this is fixed in a separate commit as this/was is a problem in existing code.